core/module_specifier.rs

```diff
@@ -80,3 +134,256 @@ impl PartialEq<String> for ModuleSpecifier {
     &self.to_string() == other
   }
 }
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn test_resolve_from_cwd() {
+    // Assuming cwd is the deno repo root.
+    let cwd = std::env::current_dir().unwrap();
+    let cwd_string = String::from(cwd.to_str().unwrap()) + "/";
+
+    if cfg!(target_os = "windows") {
+      let expected_url = "file:///C:/deno/tests/006_url_imports.ts";
+
+      assert_eq!(
+        ModuleSpecifier::resolve_from_cwd(r"C:/deno/tests/006_url_imports.ts")
+          .unwrap()
+          .to_string(),
+        expected_url,
+      );
+      assert_eq!(
+        ModuleSpecifier::resolve_from_cwd(r"C:\deno\tests\006_url_imports.ts")
+          .unwrap()
+          .to_string(),
+        expected_url,
+      );
+      assert_eq!(
+        ModuleSpecifier::resolve_from_cwd(r"/deno/tests/006_url_imports.ts")
+          .unwrap()
+          .to_string(),
+        format!(
+          "file:///{}",
+          cwd.join("/deno/tests/006_url_imports.ts").to_str().unwrap(),
+        ).replace("\\", "/")
+      );
+      assert_eq!(
+        ModuleSpecifier::resolve_from_cwd(r"\tests\006_url_imports.ts")
+          .unwrap()
+          .to_string(),
+        format!(
+          "file:///{}",
+          cwd.join(r"\tests\006_url_imports.ts").to_str().unwrap(),
+        ).replace("\\", "/")
+      );
+
+    // TODO: make those tests pass
+    // non-disk paths
+    //      let invalid_paths = vec![
```

Comment:
I'll need a bit of help with those cases, @piscisaureus could you take a look?

Reply:
Were these passing before? Is this a blocker to landing this patch?

Reply:
They didn't exist before this PR 😅@piscisaureus suggested them [in this comment](https://github.com/denoland/deno/pull/2581#issuecomment-506059402).